Geometry_Engine/Query/Area.cs

```diff
@@ -43,6 +43,12 @@ public static partial class Query
         [PreviousVersion("4.3", "BH.Engine.Geometry.Query.IArea(BH.oM.Geometry.IGeometry)")]
         public static double IArea(this IGeometry geometry, double tolerance = Tolerance.Distance)
         {
+            if (geometry == null)
+            {
+                BH.Engine.Reflection.Compute.RecordError("Cannot query the area of null geometry.");
+                return double.NaN;
```

Comment:
Shouldn't we return 0 in such cases, @FraserGreenroyd? I do not remember what was the final decision about it?

Reply:
0 could be a valid number for downstream operations, whereas `double.NaN` prevents downstream operations carrying on as if everything was ok with just the error recorded (particularly if used in code calling the method rather than just UI components). Final decision is it needs to make sense at the time of being done, some instances 0 or -1 are acceptable, others are `double.NaN`. My preference is on `double.NaN` but they obviously then need checking for in code calling them, but that's just good practice to garbage check anything coming back.